pkg/gitops/flux/flux.go

```diff
@@ -149,6 +151,78 @@ func (f *Flux) Uninstall(ctx context.Context, cluster *types.Cluster, clusterSpe
 	return nil
 }
 
+func (f *Flux) PauseClusterResourcesReconcile(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec, provider providers.Provider) error {
+	if f.shouldSkipFlux() {
+		logger.Info("GitOps field not specified, pause cluster resources reconcile skipped")
+		return nil
+	}
+
+	c, err := f.fluxClient.GetCluster(ctx, cluster, clusterSpec)
+	if err != nil {
+		return err
+	}
+	if c.Spec.GitOpsRef == nil {
+		logger.Info("GitOps not enabled in the existing cluster, pause cluster resources reconcile skipped")
+		return nil
+	}
+
+	logger.V(3).Info("Pause Flux EKS-A resources reconcile")
+
+	if err := f.fluxClient.DisableResourceReconcile(ctx, cluster, clusterSpec.Cluster.ResourceType(), clusterSpec.Cluster.Name, clusterSpec.Cluster.Namespace); err != nil {
+		return fmt.Errorf("disable resource %s %s from Flux reconcile: %v", clusterSpec.Cluster.ResourceType(), clusterSpec.Cluster.Name, err)
+	}
+
+	if err := f.fluxClient.DisableResourceReconcile(ctx, cluster, provider.DatacenterResourceType(), clusterSpec.Cluster.Spec.DatacenterRef.Name, clusterSpec.Cluster.Namespace); err != nil {
+		return fmt.Errorf("disable resource %s %s from Flux reconcile: %v", provider.DatacenterResourceType(), clusterSpec.Cluster.Spec.DatacenterRef.Name, err)
+	}
+
+	if provider.MachineResourceType() != "" {
+		for _, machineConfigRef := range clusterSpec.Cluster.MachineConfigRefs() {
+			if err := f.fluxClient.DisableResourceReconcile(ctx, cluster, provider.MachineResourceType(), machineConfigRef.Name, clusterSpec.Cluster.Namespace); err != nil {
+				return fmt.Errorf("disable resource %s %s from Flux reconcile: %v", provider.MachineResourceType(), machineConfigRef.Name, err)
+			}
+		}
+	}
```

Comment:
should we include here the rest of the objects, like OIDC, AWSIam, etc?

Reply:
i prefer to create issue to follow up, since it is not just in flux, but same for cluster_manager add pause annotation.

Reply:
https://github.com/aws/eks-anywhere/issues/3127